requirements/Makefile

```diff
@@ -7,6 +7,8 @@ requirements:
 	pip-compile -o test-requirements.txt test-requirements.in --allow-unsafe
 	pip-compile -o dev-requirements.txt dev-requirements.in --allow-unsafe
 	pip-compile -o docs-requirements.txt docs-requirements.in --allow-unsafe
+	awk 'BEGIN { del=0 } /appnope/ { del=1 } del<=0 { print } /via ipython/ { del -= 1 }' prod-requirements.txt > temp-prod-requirements.txt && mv temp-prod-requirements.txt prod-requirements.txt
+	awk 'BEGIN { del=0 } /appnope/ { del=1 } del<=0 { print } /via ipython/ { del -= 1 }' dev-requirements.txt > temp-dev-requirements.txt && mv temp-dev-requirements.txt dev-requirements.txt
```

Comment:
Some quick testing showed requirements file corruption if the `appnope` text showed up in any way other than exactly:

```
appnope==x.y.z
    # via ipython
```
I think `sed` might be a bit safer (plus has the `-i` (in-place) option, which allows dropping the `&& mv ...`). The following matches the exact text _only_:
```bash
sed -E -i 'N;/^appnope==.+\n    # via ipython$/d' dev-requirements.txt
```
Where:
- `-E` == use extended regex
- `-i` == edit file in-place
- `'N;` == join next input line to the pattern space
- `/d` == delete the matched text

Reply:
thanks. I'm really not familiar with either `awk` or `sed` at all, so this is all new to me

Reply:
huh I think this is why i went with `awk` instead of `sed`
```
# sed -E -i 'N;/^appnope==[^\n]+\n    # via ipython$/d' prod-requirements.txt
sed: 1: "prod-requirements.txt": extra characters at the end of p command
```

Reply:
looks like it's a mac os thing https://stackoverflow.com/questions/16745988/sed-command-with-i-option-in-place-editing-works-fine-on-ubuntu-but-not-mac

Reply:
huh, I've fixed it, but it looks like `sed` is also deleting lines that have 
```
    #   zope-interface
```
in them, even though the regex definitely would not match this. any ideas? @joxl

Reply:
> looks like it's a mac os thing

Ugh yes, BSD sed is awful (it insists upon making the argument following `-i` the "backup extension", unless it recognizes the argument (e.g. `-i -e`). This version is compatible on both BSD and GNU `sed`:
```bash
sed -E -i -e 'N;/^appnope==[^\n]+\n    # via ipython$/d' prod-requirements.txt
```

Reply:
> even though the regex definitely would not match this. any ideas?

Huh, definitely a head-scratcher. The only thing I can think of is make sure the `Makefile` version has the `$` escaped. That is:
- `sed -E -i -e 'N;/^appnope==[^\n]+\n    # via ipython$/d' ...` (Bash)
- `sed -E -i -e 'N;/^appnope==[^\n]+\n    # via ipython$$/d' ...` (Makefile)